blockly-ws-search/src/WorkspaceSearch.js

```diff
@@ -0,0 +1,660 @@
+    this.inputElement_ = this.createTextInput_();
+    this.inputElement_.addEventListener('keydown',
+        evt => this.onKeyDown_(/** @type {KeyboardEvent} */ evt));
+    this.inputElement_.addEventListener('input', () => this.onInput_());
+    this.inputElement_.addEventListener('click',
+        () => this.searchAndHighlight(this.preserveSelected));
+    inputWrapper.append(this.inputElement_);
+    searchContent.append(inputWrapper);
+
+    this.actionDiv_ = document.createElement('div');
+    Blockly.utils.dom.addClass(this.actionDiv_, 'ws-search-actions');
+    searchContent.append(this.actionDiv_);
+
+    const nextBtn = this.createNextBtn_();
+    if (nextBtn) {
+      this.addActionBtn(nextBtn, () => this.next());
+    }
+
+    const previousBtn = this.createPreviousBtn_();
+    if (previousBtn) {
+      this.addActionBtn(previousBtn, () => this.previous());
+    }
+
+    const closeBtn = this.createCloseBtn_();
+    if (closeBtn) {
+      this.addBtnListener_(closeBtn, () => this.close())
+      searchContainer.append(closeBtn);
+    }
+
+    this.htmlDiv_.append(searchContainer);
+
+    parentSvg.parentNode.insertBefore(this.htmlDiv_, parentSvg);
+  }
+
+  /**
+   * Add a button to the action div. This must be called after the init function
+   * has been called.
+   * @param {!HTMLButtonElement} btn The button to add the event listener to.
+   * @param {!Function} onClickFn The function to call when the user clicks on 
+   *     or hits enter on the button.
+   */
+  addActionBtn(btn, onClickFn) {
+    this.addBtnListener_(btn, onClickFn);
+    this.actionDiv_.append(btn);
+  }
+
+  /**
+   * Creates the text input for the search bar.
+   * @return {!HTMLInputElement} A text input for the search bar.
+   * @protected
+   */
+  createTextInput_() {
+    let textInput = document.createElement('input');
+    textInput.type = 'text';
+    textInput.setAttribute('placeholder', this.textInputPlaceholder_);
+    return textInput;
+  }
+
+  /**
+   * Creates the button used to get the next block in the list.
+   * @return {!HTMLButtonElement} The next button.
+   * @protected
+   */
+  createNextBtn_() {
+    return this.createBtn('next-btn', 'Find next');
+  }
+
+  /**
+   * Creates the button used to get the previous block in the list.
+   * @return {!HTMLButtonElement} The previous button.
+   * @protected
+   */
+  createPreviousBtn_() {
+    return this.createBtn('previous-btn', 'Find previous');
+  }
+
+  /**
+   * Creates the button used for closing the search bar.
+   * @return {!HTMLElement} A button for closing the search bar.
+   * @protected
+   */
+  createCloseBtn_() {
+    return this.createBtn('close-btn', 'Close search bar');
+  }
+
+  /**
+   * Creates a button for the workspace search bar.
+   * @param {string} className The class name for the button.
+   * @param {string} text The text to display to the screen reader.
+   * @return {!HTMLButtonElement} The created button.
+   */
+  createBtn(className, text) {
```

Comment:
Why is this public?

Reply:
Because this adds some nice properties for accessibility. If we are giving users the option to addActionBtn then I also wanted to give them an easy way to create an accessible button.

Reply:
I think making this public is scope creep. Developers can create a button, set a class name and add an aria-label themselves, we don't need to support that functionality.

Reply:
ok done.